aws/resource_aws_lambda_function.go

```diff
@@ -47,22 +48,57 @@ func resourceAwsLambdaFunction() *schema.Resource {
 			"filename": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"s3_bucket", "s3_key", "s3_object_version"},
+				ConflictsWith: []string{"s3_bucket", "s3_key", "s3_object_version", "image_uri"},
 			},
 			"s3_bucket": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"filename"},
+				ConflictsWith: []string{"filename", "image_uri"},
 			},
 			"s3_key": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"filename"},
+				ConflictsWith: []string{"filename", "image_uri"},
 			},
 			"s3_object_version": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"filename"},
+				ConflictsWith: []string{"filename", "image_uri"},
+			},
+			"image_uri": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"filename", "s3_bucket", "s3_key", "s3_object_version"},
+			},
+			"package_type": {
```

Comment:
Can't this just be computed based on s3 / code or image?

Reply:
I mean, you can now end up in a situation where a user will choose a ZIP and an image which isn't correct

Reply:
(I know there's a validator but why even need it)

Reply:
What if AWS adds support for another archive type later on? im assuming the API was designed to be open for expansion in the future, we should follow the API. https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/ and such...

Reply:
Good catch. We typically don't automatically set parameters like this. We should, however, add validation along the lines of:
* if `package_type` is `Image`, `image_uri` must be set and none of `filename` or `s3_*` can be set
* if `package_type` is `Zip`, `image_uri` and `image_config` cannot be set and either `filename` or some combination of `s3_*` must be set